services/user.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.auth import get_user_model
+
+
+def create_user(username: str,
+                password: str,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+    user_model = get_user_model()
+    user = user_model(username=username)
+    user.set_password(password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> int:
+    return get_user_model().objects.get(id=user_id)
+
+
+def update_user(user_id: int,
+                username: str = None,
+                password: str = None,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+
+    user = get_user_model().objects.get(id=user_id)
+
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user

```

Comment:
The same issue is present in the `update_user` function. You're calling `user.save()` even if no changes were made to the user. To improve the performance of your code, only call `save()` if at least one field was updated. You can use a boolean variable to track whether any field was updated. If this variable is `True` at the end of the function, call `save()`. Otherwise, don't call it.